src/nodes.coffee

```diff
@@ -4779,15 +4791,8 @@ exports.StringWithInterpolations = class StringWithInterpolations extends Base
     fragments.push @makeCode '`' unless @jsx
     for element in elements
       if element instanceof StringLiteral
-        element.value = element.unquote yes, @jsx
-        unless @jsx
-          # Backticks and `${` inside template literals must be escaped.
-          element.value = element.value.replace /(\\*)(`|\$\{)/g, (match, backslashes, toBeEscaped) ->
-            if backslashes.length % 2 is 0
-              "#{backslashes}\\#{toBeEscaped}"
-            else
-              match
-        fragments.push element.compileToFragments(o)...
+        unquotedElementValue = if @jsx then element.unquotedValueForJSX else element.unquotedValueForTemplateLiteral
+        fragments.push @makeCode unquotedElementValue
```

Comment:
So here when compiling a `StringWithInterpolations`'s text chunks:

Rather than mutating the `StringLiteral`'s `value` and then asking it to compile itself, we're taking full control over its compilation and just plucking its unquoted value that we need